src/cookie/cookie.ts

```diff
@@ -0,0 +1,206 @@
+      this.consent = this.hasConsentForAnalytics();
+      this.hideElement(this.cookieBannerContainer[0] as HTMLElement);
+    } else {
+      //add event listerners
+      this.acceptCookies[0]?.addEventListener(
+        "click",
+        this.handleAcceptClickEvent.bind(this),
+      );
+      this.rejectCookies[0]?.addEventListener(
+        "click",
+        this.handleRejectClickEvent.bind(this),
+      );
+
+      const hideButtons = Array.prototype.slice.call(this.hideCookieBanner);
+      for (let i = 0; i < hideButtons.length; i++) {
+        hideButtons[i].addEventListener(
+          "click",
+          this.handleHideButtonClickEvent.bind(this),
+        );
+      }
+    }
+  }
+
+  /**
+   * Handles the click event when the accept button is clicked.
+   *
+   * @param {Event} event - The click event.
+   */
+  handleAcceptClickEvent(event: Event): void {
+    event.preventDefault();
+    this.setBannerCookieConsent(true, window.location.hostname);
+    this.consent = true;
+    window.DI.analyticsGa4.loadGtmScript();
+  }
+
+  /**
+   * Handles the click event when the reject button is clicked.
+   *
+   * @param {Event} event - The click event object.
+   */
+  handleRejectClickEvent(event: Event): void {
+    event.preventDefault();
+    this.consent = false;
+    this.setBannerCookieConsent(false, window.location.hostname);
+  }
+
+  /**
+   * Handles the click event of the hide button.
+   *
+   * @param {Event} event - The click event.
+   */
+  handleHideButtonClickEvent(event: Event): void {
+    event.preventDefault();
+    this.hideElement(this.cookieBannerContainer[0] as HTMLElement);
+  }
+
+  /**
+   * Set the banner cookie consent.
+   *
+   * @param {boolean} analyticsConsent - Whether analytics consent is given or not.
+   * @param {string} domain - The domain where the cookie is set.
+   */
+  setBannerCookieConsent(analyticsConsent: boolean, domain: string): void {
+    this.setCookie(
+      this.COOKIES_PREFERENCES_SET,
+      { analytics: analyticsConsent },
+      { days: 365 },
+      domain,
+    );
+
+    if (this.cookieBanner) {
+      this.hideElement(this.cookieBanner);
+    }
+
+    if (analyticsConsent === true) {
+      if (this.cookiesAccepted) {
+        this.showElement(this.cookiesAccepted);
+      }
+
+      let event;
+      if (typeof window.CustomEvent === "function") {
+        event = new window.CustomEvent("cookie-consent");
+      } else {
+        event = document.createEvent("CustomEvent");
+        event.initCustomEvent("cookie-consent");
+      }
+      window.dispatchEvent(event);
+    } else {
+      if (this.cookiesRejected) {
+        this.showElement(this.cookiesRejected);
+      }
+    }
+  }
+
+  /**
+   * Checks whether the user has given consent for analytics.
+   *
+   * @return {boolean} - Returns true if the user has given consent for analytics, false otherwise.
+   */
+  hasConsentForAnalytics(): boolean {
+    const cookieConsent = JSON.parse(
```

Comment:
The cookie is always a string.
In this case, we are using the object JSON because we want to access to the method "parse" and transform the string we saved in the cookie into a nice object we can manipulate.
More docs:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse